SAP/spartacus
# fix: CXSPA-8359 unknown HTTP error printed twice to the console

This PR contains a fix for the issue when the same unknown HTTP error was printed to the console twice - by `HttpErrorHandlingInterceptor` and by `UnknownErrorHandler`. 
![image](https://github.com/user-attachments/assets/14a20b61-a8d0-4a63-b9b1-4c27461f27f1)

To solve the issue, the logic responsible for printing errors in `UnknownErrorHandlers` was adjusted and wrapped by the `ssrStrictErrorHandlingForHttpAndNgrx` feature toggle. After adjustment, additional logs from this error handler will be present only in dev mode (both, CSR and SSR), so production logs will be free from duplication.

QA steps:
- go to `UnknownErrorHandler` and change the return value from `getPriority` to `Priority.HIGH` (to ensure the error handler is going to be triggered)
- mock HTTP error by changing `default-cms-config.ts`:
   ```
   export const defaultCmsModuleConfig: CmsConfig = {
    backend: {
      occ: {
        endpoints: {
          /* ... */
          components: 'cms/components2',
        },
      },
    },
   };
   ```
- run `npm build`, `npm run build:ssr` and `npm run serve:ssr:dev` and verify whether HTTP error `Http failure response for ...` is presented once
- run `npm run start` and verify whether HTTP error `Http failure response for ...` is presented twice
   - with the prefix: `An unknown http error occured`
   - wrapped by `OutboundHttpError: Outbound HTTP Error`
- run `npm run dev:ssr` and verify whether HTTP error `Http failure response for ...` is presented twice
   - with the prefix: `An unknown http error occured`
   - wrapped by `OutboundHttpError: Outbound HTTP Error`

closes [CXSPA-8359](https://jira.tools.sap/browse/CXSPA-8359)
